### DISCUSSION TITLE
Serverless function deployment is ignoring --force flag

### DISCUSSION THREAD

**Discussion by DavidKvas** [vercel/vercel#8958] [Help] [answered]
Hi, recently, my vercel serverless function deployment stopped working. In the project settings, I have overridden the install command like this (so we can use some legacy libraries): npm install --force.

It seems that serverless function deployment is ignoring this --force flag. 

This is the error from the deployment log: https://share.cleanshot.com/0eFybM

Here are the build settings for our project: https://share.cleanshot.com/Zo3uRK

Any help would be much appreciated.

Thanks,

David

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out @DavidKvas. Do you get the same result on a local build using [`vercel build`](https://vercel.com/docs/cli/build#) or [`vercel build --debug`](https://vercel.com/docs/cli/global-options#debug)?

**Reply by DavidKvas** [reply]
It's building without issue locally. It's happening only when deploying. Please note this started happening just last week. It was working fine before.

**Reply by amyegan** [reply]
I found your support ticket with the full log details and noticed that you have a warning which says `Detected “engines”: { “node”: “>=16.14.0” } in your 'package.json' that will automatically upgrade when a new major Node.js Version is released.` 

That means your [Node.js Version](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/node-js#node.js-version) will automatically upgrade to 18 once it is available. Since [Node 18 LTS was enabled](https://vercel.com/changelog/node-js-18-lts-is-now-available) just last week, that could be the source of your problem. 

You should change the `engine.node` value to "16", or remove that property and set the Node version to 16.x in your General Project Settings. You'll need to redeploy after that to kick off a new build with the correct Node.js version.
![image](https://user-images.githubusercontent.com/8247105/203357509-8f26a668-0a3b-41eb-b776-0bc61127e91b.png)

If that doesn't solve it for you, can you pull down a fresh clone of the repo and try running `vercel build --debug` there? I'm wondering if your local build picking up packages that you already have installed in your original clone so we don't get to see the error details.

**Reply by DavidKvas** [reply]
Thanks, @amyegan. I've removed the node engine section from the package.json and upgraded to Node v18 (I'm happy to go with v18). 

Now, the warning message ("Detected engines...") is gone. However, I'm still having the same issue. Here is the screenshot from the deployment: https://share.cleanshot.com/od3NcK

It seems that the introduction of Node v18 broke the deployment of the serverless functions (it's a weird coincidence that my issue started just last week). 

Btw what did you mean by "a fresh clone of the repo"?

**Reply by DavidKvas** [reply]
@amyegan  One more thing. From the deployment screenshot, it's clear that **--force** flag is being removed from **npm install** when the serverless function is deployed. This seems to be a bug in the vercel deployment process. Can you verify it?

**Reply by amyegan** [reply]
Can you set it to v16 and try again? I think that v18 might be incompatible with some of your package versions. I want to make sure we rule that out before we try more complicated solutions

**Reply by DavidKvas** [reply]
Yep, that fixed it. It seems it's not working only for Node v18, which is fine for now, as we can stay on v16 for the time being.

Thanks for your help, really appreciate it.
